crates/agent-sql/src/id.rs

```diff
@@ -5,6 +5,17 @@ use sqlx::{postgres, Decode, Encode, Type, TypeInfo};
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct Id([u8; 8]);
```

Comment:
As it is now, we still need separate copies of `Id` and `CatalogType` because `models` doesn't depend on `sqlx` and thus can't `impl sqlx::Type`. I'm half convinced that we should add an optional `sqlx` dependency (and feature) to `models` in order to avoid this duplication and having conversions sprinkled everywhere, but figured we could address it in a subsequent PR.